core/src/main/java/jeeves/transaction/TransactionManager.java

```diff
@@ -72,6 +74,15 @@
 
             result = action.doInTransaction(transaction);
 
+            // GlobalExceptionManager stores the exception in a request attribute, to be processed to rollback the transaction
+            // as otherwise GlobalExceptionManager processes the exception and the code doesn't enter in the catch block
+            // to rollback the transaction
+            Throwable ex = (Throwable) RequestContextHolder.currentRequestAttributes().getAttribute("exception", RequestAttributes.SCOPE_REQUEST);
+
+            if (ex != null) {
```

Comment:
Nicely done:
- variable names matter, `ex` —> `requestException` (Or anything more descriptive)
- I would place this check in a finally block to communicate it is part of exception handling and not the normal course of logic